### ISSUE TITLE
Don't allow duplicate coupling schemes in compositional coupling scheme

### ISSUE THREAD

**Issue by BenjaminRodenberg** [precice/precice#1695]
The following compositional coupling scheme is currently allowed:
```
  <coupling-scheme:parallel-implicit>
    <participants first="SolverOne" second="SolverTwo" />
    ...
  </coupling-scheme:parallel-implicit>

  <coupling-scheme:parallel-explicit>
    <participants first="SolverOne" second="SolverThree" />
    ...
  </coupling-scheme:parallel-explicit>

  <coupling-scheme:parallel-explicit>
    <participants first="SolverOne" second="SolverThree" />
    ...
  </coupling-scheme:parallel-explicit>
</precice-configuration>
```

I think one of the two parallel explicit schemes is not needed and we should raise an error. We could easily check for duplicates of coupling schemes between two identical solvers.

Again: If we reduce the number of legal configurations, this will reduce the number of required tests, simplify implementation and make preCICE more user-friendly.

**Comment by uekerman** [comment]
For me, this is clearly a bug and not a usability concern. This has never been a legal configuration and would surely lead to a crash somewhere. Adding a check for duplicated coupling schemes should not be too complicated.

**Comment by fsimonis** [comment]
This check should be simple to implement using `CouplingScheme::getCouplingPartners()` when composing schemes here:

https://github.com/precice/precice/blob/d1d4097fa5992fbc4d63092cf3c3020e8e702816/src/cplscheme/CompositionalCouplingScheme.cpp#L28

One would need to check that `getCouplingSchemes()` of the compositional scheme doesn't include CouplingPartners of the scheme to add.

**Comment by Tanisha127** [comment]
Hi @BenjaminRodenberg and @fsimonis 
I would like to work on this issue.
From the discussion, I understand that the duplicate check should be implemented in CompositionalCouplingScheme::addCouplingScheme() by comparing the coupling partners using getCouplingPartners().
Please let me know if the issue is still available and if there are any additional constraints I should consider before starting.
Thanks!
